lib/web_ui/lib/src/engine/web_experiments.dart

```diff
@@ -0,0 +1,43 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+// @dart = 2.6
+part of engine;
+
+final WebExperiments webExperiments = WebExperiments._();
+
+/// A bag of all experiment flags in the web engine.
+///
+/// This class also handles platform messages that can be sent to enable/disable
+/// certain experiments at runtime without the need to access engine internals.
+class WebExperiments {
+  WebExperiments._() {
+    js.context['_flutter_internal_update_experiment'] = updateExperiment;
+  }
+
+  /// Experiment flag for using canvas-based text measurement.
+  bool get useCanvasText => _useCanvasText ?? false;
```

Comment:
Does allowing `null` help with anything?

Reply:
Yes. In tests, we set the flag to a specific value (true or false), then we set it back to `null` when the test is done. The value `null` means "use the default value".